### PR TITLE
Table Keyboard Navigation HLD

### PR DESCRIPTION
**Pull request by atmgrifter00**
# Pull Request

## 🤨 Rationale

This is an HLD to discuss implementation strategies related to keyboard navigation.

- #1137
- #1151

### CODE REVIEW

File: packages/nimble-components/src/table/specs/table-keyboard-navigation-hld.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    - Moving focus to one of multiple interactive elements in a cell requires either pressing `Tab` or pressing `Enter` on the cell if it is focused.
+
+    Pros:
+
+    - No reliance on modifier keys to navigate between the cell content and to expand/collapse rows.
+    - Provides expected screen reader behaviors for announcing content cell-by-cell.
+
+    Cons:
+
+    - Screen readers will now only announce contents of a row when the row is highlighted.
```
[the remark this answers is not in the document]
Comment (atmgrifter00) [reply packages/nimble-components/src/table/specs/table-keyboard-navigation-hld.md]: I have no idea what I meant. I've changed this to a '?', as I'm not sure I could come up with a valid con.

File: packages/nimble-components/src/table/specs/table-keyboard-navigation-hld.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+2. Pressing `Tab` will focus first row, followed by all interactive elements in the row
+
+    Pros:
+
+    - Allows for quick navigation through the row's contents
+    - Provides a less laborious means of focusing one of multiple interactive elements in a single cell.
+
+    Cons:
+
+    - The user no longer has a means of tabbing "quickly" past the table. They must tab through the interactive elements of a single row first. This seems like a marginal con, as a row typically wouldn't have enough interactive content for this to be too cumbersome.
```
[the remark this answers is not in the document]
Comment (atmgrifter00) [reply packages/nimble-components/src/table/specs/table-keyboard-navigation-hld.md:128]: I've added some more text in the '2.' section that describes what happens when `Tab` is pressed after reaching the last cell.

File: packages/nimble-components/src/table/specs/table-keyboard-navigation-hld.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+2. The `Tab` key moves focus between the focusable elements in a row (not the cells). When focus has reached the end (or beginning of the row), a continued `Tab` (or `Shift-Tab`) will move focus outside of the table.
+3. When a cell has multiple interactive elements, a user _must_ press `Enter` to move the focus to the first interactive element in the cell. At this point the table is no longer in navigation mode, and a user would now press `Tab` to move to other interactive elements within the same cell. _Disclaimer_: this guidance applies to any `grid` or `treegrid` pattern.
+
+If the above guidelines are adhered to, technologies like screen readers should be able to handle navigating through the table in an expected fashion.
+
+### Option 1: Diverge from ARIA guidelines
+
+If you consult the IxD document you will see that the design diverges from the above ARIA guidance in the follwing ways:
```
[the remark this answers is not in the document]
Comment (atmgrifter00) [reply packages/nimble-components/src/table/specs/table-keyboard-navigation-hld.md]: I've updated the document to call this out specifically.

**Review by leslieab — APPROVED** [review]

**Comment by leslieab** [comment]
We should go with **Option 2: Stick to ARIA guidelines**, as there are a lot less cons in doing this than deviating from it.

File: change/@ni-nimble-components-a062b0e0-e618-4df7-88d5-43aad835cf68.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (rajsite) [inline change/@ni-nimble-components-a062b0e0-e618-4df7-88d5-43aad835cf68.json:1]: @atmgrifter00 I think this was just put on the backburner so I'll move it back to draft.

**Comment by rajsite** [comment]
@atmgrifter00 closing this in favor of #1960, I think that PR covers everything but an fyi just in case

**Pull request closed without merging**